idea/src/org/jetbrains/kotlin/idea/configuration/KotlinMavenArchetypesProvider.kt

```diff
@@ -57,11 +58,10 @@ class KotlinMavenArchetypesProvider : MavenArchetypesProvider {
             .asJsonArray
             .map { it.asJsonObject }
             .map { MavenArchetype(it.get("g").asString, it.get("a").asString, it.get("v").asString, null, null) }
-            .filter { it.version.matches(versionRegexp) }
-            .groupBy { it.artifactId + "-" + it.version.substringBeforeLast('.') }
-            .mapValues { it.value.maxBy { a -> a.version.substringAfterLast('.').toInt() }!! }
-            .values
-            .toList()
+            .filter { it.version?.startsWith(versionPrefix) ?: false }
+            .maxBy { MavenVersionComparable(it.version) }
+            ?.let { listOf(it) }
```

Comment:
It might be cleaner to use `listOfNotNull` instead of `?.let { listOf } ?: emptyList()`